src/core/core.controller.js

```diff
@@ -605,6 +605,11 @@ module.exports = function(Chart) {
 		getDatasetMeta: function(datasetIndex) {
 			var me = this;
 			var dataset = me.data.datasets[datasetIndex];
+			
+			if (typeof dataset == "undefined"){
```

Comment:
I see, I'll try to make a demo on JSFiddle, the calling method causing the error is updateHoverStyle.

```
updateHoverStyle: function(elements, mode, enabled) {
			var method = enabled? 'setHoverStyle' : 'removeHoverStyle';
			var element, i, ilen;

			for (i=0, ilen=elements.length; i<ilen; ++i) {
				element = elements[i];
				if (element) {
					try {
						this.getDatasetMeta(element._datasetIndex).controller[method](element);
					} catch (e) {
						// Supress errors
					}
				}
			}
		},
```